### PR TITLE
Guard against nullptr in ParseNumber()

### PR DESCRIPTION
**Pull request by Qining**
Returns false if the given string is as nullptr.

### CODE REVIEW

File: source/util/parse_number.h [hunk 1]
```diff
@@ -171,6 +171,7 @@ bool ParseNumber(const char* text, T* value_pointer) {
   static_assert(sizeof(T) > 1,
                 "Don't use a single-byte type this parse method");
```
Comment (antiagainst) [inline source/util/parse_number.h]: The error message is strange.

Maybe
Single-byte types not supported in this function?

**Comment by antiagainst** [comment]
+2. Just need to fix a weird error message.

**Comment by Qining** [comment]
Guards against nullptr are also added to ParseAndEncodeXXX() functions.
Tests are also added.

**Comment by antiagainst** [comment]
+2

**Pull request merged**